Revert our changes and use FireMasterK's improvements

Stypox opened the pull request:
This PR does these things:
- it reverts all of our commits related to parsing custom JSON format. You can see all of our commits from [here](https://github.com/mmastrac/nanojson/compare/master...TeamNewPipe:nanojson:master):
  * | 72c8ed4 Revert "Support JS-like JSON"
  * | 41f29d8 Revert "Support keys starting with a non-ASCII character"
  * | d3b1b70 Revert "Support single quote strings"
  * | f9d8d68 Revert "Support keywords as keys"
  * | e2cd3f0 Revert "Fix some more semi-string parsing bugs"
  * | 714bb89 Revert "Disallow semi-strings as value"
  * | b52e21a Revert "Fix Checkstyle errors"
  * | 9718d33 Revert "Fix issue for semi-strings as well"
- the only two commits that were not reverted are 0e64c2b7d1012b440c6e066539f4d2794304e989  d5da581ec79e6cc12962da8b92749038143f8e08, which are both also present in FireMasterK's fork
- additionally, the README changes were retained (5494f3c76084e7693efd56daf1e8db8f9284ffbe) and updated with fcb2efdc0192bb8c39b120283eedeb13d1d4a5b6
- pulls in updates from upstream
- pulls in @FireMasterK's [performance changes](https://github.com/FireMasterK/nanojson/commits/master/) (note that this required reverting the other commits, since @FireMasterK's fork has just removed those commits directly, but I would prefer not to force-push on our `master` branch).
- I skipped commit c624b79f66757d0ce0c59c50ae97786905fca070 since it pulls in the `fastutil` library to squeeze out the last bits of performance, but we'd prefer not to increase APK size, as suggested by @FireMasterK in https://github.com/TeamNewPipe/NewPipeExtractor/pull/981#discussion_r1031664328, also see [TeamNewPipe/NewPipeExtractor@`325c07d` (#981)](https://github.com/TeamNewPipe/NewPipeExtractor/pull/981/commits/325c07def9a4ced4a7900b7a4d58fb4db7d97003)
- I did include commit 24eb3368a2517299db57871601ad09996db11a99 though, even though it adds the `fastdoubleparser` library. I don't think it matters much whether we keep it or not, the JAR is just [191kB](https://mvnrepository.com/artifact/ch.randelshofer/fastdoubleparser/2.0.1) but I'm pretty sure most of it would be optimized out or merged during build. Let me know if I should revert this.
- I also skipped commit a507525e549a836c3a8b6ab7090dca38e92942ef and instead made sure we retain compatibility with JDK 11 with 2a27a1787eb2f60dfe8cbe3168bef3649c934884.
- fixes checkstyle

All tests pass. This PR is basically a replacement for https://github.com/TeamNewPipe/NewPipeExtractor/pull/981, where many people suggested updating our own fork. While I initially was against, I now think it's necessary if we want to have some of @FireMasterK's commits but not all.

Closes #2

dependabot[bot] committed:
Bump maven-javadoc-plugin from 3.3.0 to 3.3.2

Bumps [maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.3.0 to 3.3.2.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.3.0...maven-javadoc-plugin-3.3.2)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump maven-site-plugin from 3.9.1 to 3.11.0

Bumps [maven-site-plugin](https://github.com/apache/maven-site-plugin) from 3.9.1 to 3.11.0.
- [Release notes](https://github.com/apache/maven-site-plugin/releases)
- [Commits](https://github.com/apache/maven-site-plugin/compare/maven-site-plugin-3.9.1...maven-site-plugin-3.11.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-site-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump nexus-staging-maven-plugin from 1.6.8 to 1.6.12

Bumps nexus-staging-maven-plugin from 1.6.8 to 1.6.12.

---
updated-dependencies:
- dependency-name: org.sonatype.plugins:nexus-staging-maven-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump maven-compiler-plugin from 3.8.1 to 3.10.1

Bumps [maven-compiler-plugin](https://github.com/apache/maven-compiler-plugin) from 3.8.1 to 3.10.1.
- [Release notes](https://github.com/apache/maven-compiler-plugin/releases)
- [Commits](https://github.com/apache/maven-compiler-plugin/compare/maven-compiler-plugin-3.8.1...maven-compiler-plugin-3.10.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-compiler-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #67 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-compiler-plugin-3.10.1

Bump maven-compiler-plugin from 3.8.1 to 3.10.1

mmastrac committed:
Merge pull request #61 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-javadoc-plugin-3.3.2

Bump maven-javadoc-plugin from 3.3.0 to 3.3.2

mmastrac committed:
Merge pull request #65 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-site-plugin-3.11.0

Bump maven-site-plugin from 3.9.1 to 3.11.0

mmastrac committed:
Merge pull request #66 from mmastrac/dependabot/maven/org.sonatype.plugins-nexus-staging-maven-plugin-1.6.12

Bump nexus-staging-maven-plugin from 1.6.8 to 1.6.12

dependabot[bot] committed:
Bump maven-javadoc-plugin from 3.3.2 to 3.4.0

Bumps [maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.3.2 to 3.4.0.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.3.2...maven-javadoc-plugin-3.4.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump maven-site-plugin from 3.11.0 to 3.12.0

Bumps [maven-site-plugin](https://github.com/apache/maven-site-plugin) from 3.11.0 to 3.12.0.
- [Release notes](https://github.com/apache/maven-site-plugin/releases)
- [Commits](https://github.com/apache/maven-site-plugin/compare/maven-site-plugin-3.11.0...maven-site-plugin-3.12.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-site-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump nexus-staging-maven-plugin from 1.6.12 to 1.6.13

Bumps nexus-staging-maven-plugin from 1.6.12 to 1.6.13.

---
updated-dependencies:
- dependency-name: org.sonatype.plugins:nexus-staging-maven-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump exec-maven-plugin from 3.0.0 to 3.1.0

Bumps [exec-maven-plugin](https://github.com/mojohaus/exec-maven-plugin) from 3.0.0 to 3.1.0.
- [Release notes](https://github.com/mojohaus/exec-maven-plugin/releases)
- [Commits](https://github.com/mojohaus/exec-maven-plugin/compare/exec-maven-plugin-3.0.0...exec-maven-plugin-3.1.0)

---
updated-dependencies:
- dependency-name: org.codehaus.mojo:exec-maven-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #72 from mmastrac/dependabot/maven/org.codehaus.mojo-exec-maven-plugin-3.1.0

Bump exec-maven-plugin from 3.0.0 to 3.1.0

mmastrac committed:
Merge pull request #71 from mmastrac/dependabot/maven/org.sonatype.plugins-nexus-staging-maven-plugin-1.6.13

Bump nexus-staging-maven-plugin from 1.6.12 to 1.6.13

mmastrac committed:
Merge pull request #69 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-javadoc-plugin-3.4.0

Bump maven-javadoc-plugin from 3.3.2 to 3.4.0

mmastrac committed:
Merge pull request #70 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-site-plugin-3.12.0

Bump maven-site-plugin from 3.11.0 to 3.12.0

dependabot[bot] committed:
Bump maven-site-plugin from 3.12.0 to 3.12.1

Bumps [maven-site-plugin](https://github.com/apache/maven-site-plugin) from 3.12.0 to 3.12.1.
- [Release notes](https://github.com/apache/maven-site-plugin/releases)
- [Commits](https://github.com/apache/maven-site-plugin/compare/maven-site-plugin-3.12.0...maven-site-plugin-3.12.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-site-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Implements toString on JsonLazyNumber.

mmastrac committed:
Merge pull request #74 from adriantodt/issue/nanojson-68

Implements toString on JsonLazyNumber.

mmastrac committed:
Merge pull request #73 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-site-plugin-3.12.1

Bump maven-site-plugin from 3.12.0 to 3.12.1

dependabot[bot] committed:
Bump maven-javadoc-plugin from 3.4.0 to 3.4.1

Bumps [maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.4.0 to 3.4.1.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.4.0...maven-javadoc-plugin-3.4.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

FireMasterK committed:
Attempt to use CharBuffer and fastdoubleparser for better performance.

dependabot[bot] committed:
Bump maven-checkstyle-plugin from 3.1.2 to 3.2.1

Bumps [maven-checkstyle-plugin](https://github.com/apache/maven-checkstyle-plugin) from 3.1.2 to 3.2.1.
- [Release notes](https://github.com/apache/maven-checkstyle-plugin/releases)
- [Commits](https://github.com/apache/maven-checkstyle-plugin/compare/maven-checkstyle-plugin-3.1.2...maven-checkstyle-plugin-3.2.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-checkstyle-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #78 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-checkstyle-plugin-3.2.1

Bump maven-checkstyle-plugin from 3.1.2 to 3.2.1

mmastrac committed:
Merge pull request #75 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-javadoc-plugin-3.4.1

Bump maven-javadoc-plugin from 3.4.0 to 3.4.1

dependabot[bot] committed:
Bump maven-javadoc-plugin from 3.4.1 to 3.5.0

Bumps [maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.4.1 to 3.5.0.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.4.1...maven-javadoc-plugin-3.5.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

FireMasterK committed:
Attempt to lazily parse numbers by default.

dependabot[bot] committed:
Bump maven-surefire-report-plugin from 2.22.2 to 3.0.0

Bumps [maven-surefire-report-plugin](https://github.com/apache/maven-surefire) from 2.22.2 to 3.0.0.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-2.22.2...surefire-3.0.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-report-plugin
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

FireMasterK committed:
Implement lazy String parsing.

mmastrac committed:
Merge pull request #81 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-surefire-report-plugin-3.0.0

Bump maven-surefire-report-plugin from 2.22.2 to 3.0.0

mmastrac committed:
Merge pull request #79 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-javadoc-plugin-3.5.0

Bump maven-javadoc-plugin from 3.4.1 to 3.5.0

mmastrac committed:
Delete codeql-analysis.yml

mmastrac committed:
Create .github/workflows/maven.yml

mmastrac committed:
Update checkstyle.xml

FireMasterK committed:
Minimize initial buffer size to reduce memory usage.

mmastrac committed:
Bump advanced-security/maven-dependency-submission-action

mmastrac committed:
Remove dependabot step

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-surefire-report-plugin

Bumps [org.apache.maven.plugins:maven-surefire-report-plugin](https://github.com/apache/maven-surefire) from 3.0.0 to 3.2.5.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.0.0...surefire-3.2.5)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-report-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-gpg-plugin from 3.0.1 to 3.2.3

Bumps [org.apache.maven.plugins:maven-gpg-plugin](https://github.com/apache/maven-gpg-plugin) from 3.0.1 to 3.2.3.
- [Release notes](https://github.com/apache/maven-gpg-plugin/releases)
- [Commits](https://github.com/apache/maven-gpg-plugin/compare/maven-gpg-plugin-3.0.1...maven-gpg-plugin-3.2.3)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-gpg-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #90 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-surefire-report-plugin-3.2.5

Bump org.apache.maven.plugins:maven-surefire-report-plugin from 3.0.0 to 3.2.5

mmastrac committed:
Merge pull request #92 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-gpg-plugin-3.2.3

Bump org.apache.maven.plugins:maven-gpg-plugin from 3.0.1 to 3.2.3

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-checkstyle-plugin

Bumps [org.apache.maven.plugins:maven-checkstyle-plugin](https://github.com/apache/maven-checkstyle-plugin) from 3.2.1 to 3.3.1.
- [Commits](https://github.com/apache/maven-checkstyle-plugin/compare/maven-checkstyle-plugin-3.2.1...maven-checkstyle-plugin-3.3.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-checkstyle-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-source-plugin from 3.2.1 to 3.3.1

Bumps [org.apache.maven.plugins:maven-source-plugin](https://github.com/apache/maven-source-plugin) from 3.2.1 to 3.3.1.
- [Commits](https://github.com/apache/maven-source-plugin/compare/maven-source-plugin-3.2.1...maven-source-plugin-3.3.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-source-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #94 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-source-plugin-3.3.1

Bump org.apache.maven.plugins:maven-source-plugin from 3.2.1 to 3.3.1

mmastrac committed:
Merge pull request #93 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-checkstyle-plugin-3.3.1

Bump org.apache.maven.plugins:maven-checkstyle-plugin from 3.2.1 to 3.3.1

dependabot[bot] committed:
Bump org.codehaus.mojo:exec-maven-plugin from 3.1.0 to 3.2.0

Bumps [org.codehaus.mojo:exec-maven-plugin](https://github.com/mojohaus/exec-maven-plugin) from 3.1.0 to 3.2.0.
- [Release notes](https://github.com/mojohaus/exec-maven-plugin/releases)
- [Commits](https://github.com/mojohaus/exec-maven-plugin/compare/exec-maven-plugin-3.1.0...3.2.0)

---
updated-dependencies:
- dependency-name: org.codehaus.mojo:exec-maven-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.5.0 to 3.6.3

Bumps [org.apache.maven.plugins:maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.5.0 to 3.6.3.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.5.0...maven-javadoc-plugin-3.6.3)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Tillerino committed:
Add a method to JsonSink to allow separate writing of key and value

mmastrac committed:
Merge pull request #95 from mmastrac/dependabot/maven/org.codehaus.mojo-exec-maven-plugin-3.2.0

Bump org.codehaus.mojo:exec-maven-plugin from 3.1.0 to 3.2.0

mmastrac committed:
Merge pull request #96 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-javadoc-plugin-3.6.3

Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.5.0 to 3.6.3

mmastrac committed:
Merge pull request #97 from Tillerino/separate-key-value

Add a method to JsonSink to allow separate writing of key and value

mmastrac committed:
Bump version

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-gpg-plugin from 3.2.3 to 3.2.4

Bumps [org.apache.maven.plugins:maven-gpg-plugin](https://github.com/apache/maven-gpg-plugin) from 3.2.3 to 3.2.4.
- [Release notes](https://github.com/apache/maven-gpg-plugin/releases)
- [Commits](https://github.com/apache/maven-gpg-plugin/compare/maven-gpg-plugin-3.2.3...maven-gpg-plugin-3.2.4)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-gpg-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #99 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-gpg-plugin-3.2.4

Bump org.apache.maven.plugins:maven-gpg-plugin from 3.2.3 to 3.2.4

dependabot[bot] committed:
---
updated-dependencies:
- dependency-name: org.codehaus.mojo:exec-maven-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #100 from mmastrac/dependabot/maven/org.codehaus.mojo-exec-maven-plugin-3.3.0

Bump org.codehaus.mojo:exec-maven-plugin from 3.2.0 to 3.3.0

dependabot[bot] committed:
Bump org.sonatype.plugins:nexus-staging-maven-plugin

Bumps org.sonatype.plugins:nexus-staging-maven-plugin from 1.6.13 to 1.7.0.

---
updated-dependencies:
- dependency-name: org.sonatype.plugins:nexus-staging-maven-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #101 from mmastrac/dependabot/maven/org.sonatype.plugins-nexus-staging-maven-plugin-1.7.0

Bump org.sonatype.plugins:nexus-staging-maven-plugin from 1.6.13 to 1.7.0

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.6.3 to 3.7.0

Bumps [org.apache.maven.plugins:maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.6.3 to 3.7.0.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.6.3...maven-javadoc-plugin-3.7.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #102 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-javadoc-plugin-3.7.0

Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.6.3 to 3.7.0

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-checkstyle-plugin

Bumps [org.apache.maven.plugins:maven-checkstyle-plugin](https://github.com/apache/maven-checkstyle-plugin) from 3.3.1 to 3.4.0.
- [Commits](https://github.com/apache/maven-checkstyle-plugin/compare/maven-checkstyle-plugin-3.3.1...maven-checkstyle-plugin-3.4.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-checkstyle-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #103 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-checkstyle-plugin-3.4.0

Bump org.apache.maven.plugins:maven-checkstyle-plugin from 3.3.1 to 3.4.0

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-surefire-report-plugin

Bumps [org.apache.maven.plugins:maven-surefire-report-plugin](https://github.com/apache/maven-surefire) from 3.2.5 to 3.3.1.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.2.5...surefire-3.3.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-report-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.7.0 to 3.8.0

Bumps [org.apache.maven.plugins:maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.7.0 to 3.8.0.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.7.0...maven-javadoc-plugin-3.8.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #106 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-javadoc-plugin-3.8.0

Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.7.0 to 3.8.0

mmastrac committed:
Merge pull request #105 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-surefire-report-plugin-3.3.1

Bump org.apache.maven.plugins:maven-surefire-report-plugin from 3.2.5 to 3.3.1

dependabot[bot] committed:
Bump org.codehaus.mojo:exec-maven-plugin from 3.3.0 to 3.4.1

Bumps [org.codehaus.mojo:exec-maven-plugin](https://github.com/mojohaus/exec-maven-plugin) from 3.3.0 to 3.4.1.
- [Release notes](https://github.com/mojohaus/exec-maven-plugin/releases)
- [Commits](https://github.com/mojohaus/exec-maven-plugin/compare/3.3.0...3.4.1)

---
updated-dependencies:
- dependency-name: org.codehaus.mojo:exec-maven-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-gpg-plugin from 3.2.4 to 3.2.5

Bumps [org.apache.maven.plugins:maven-gpg-plugin](https://github.com/apache/maven-gpg-plugin) from 3.2.4 to 3.2.5.
- [Release notes](https://github.com/apache/maven-gpg-plugin/releases)
- [Commits](https://github.com/apache/maven-gpg-plugin/compare/maven-gpg-plugin-3.2.4...maven-gpg-plugin-3.2.5)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-gpg-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-site-plugin from 3.12.1 to 3.20.0

Bumps [org.apache.maven.plugins:maven-site-plugin](https://github.com/apache/maven-site-plugin) from 3.12.1 to 3.20.0.
- [Release notes](https://github.com/apache/maven-site-plugin/releases)
- [Commits](https://github.com/apache/maven-site-plugin/compare/maven-site-plugin-3.12.1...maven-site-plugin-3.20.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-site-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-checkstyle-plugin

Bumps [org.apache.maven.plugins:maven-checkstyle-plugin](https://github.com/apache/maven-checkstyle-plugin) from 3.4.0 to 3.5.0.
- [Commits](https://github.com/apache/maven-checkstyle-plugin/compare/maven-checkstyle-plugin-3.4.0...maven-checkstyle-plugin-3.5.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-checkstyle-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-surefire-report-plugin

Bumps [org.apache.maven.plugins:maven-surefire-report-plugin](https://github.com/apache/maven-surefire) from 3.3.1 to 3.5.0.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.3.1...surefire-3.5.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-report-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.8.0 to 3.10.0

Bumps [org.apache.maven.plugins:maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.8.0 to 3.10.0.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.8.0...maven-javadoc-plugin-3.10.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #114 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-javadoc-plugin-3.10.0

Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.8.0 to 3.10.0

mmastrac committed:
Merge pull request #113 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-surefire-report-plugin-3.5.0

Bump org.apache.maven.plugins:maven-surefire-report-plugin from 3.3.1 to 3.5.0

mmastrac committed:
Merge pull request #108 from mmastrac/dependabot/maven/org.codehaus.mojo-exec-maven-plugin-3.4.1

Bump org.codehaus.mojo:exec-maven-plugin from 3.3.0 to 3.4.1

mmastrac committed:
Merge pull request #109 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-gpg-plugin-3.2.5

Bump org.apache.maven.plugins:maven-gpg-plugin from 3.2.4 to 3.2.5

mmastrac committed:
Merge pull request #110 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-site-plugin-3.20.0

Bump org.apache.maven.plugins:maven-site-plugin from 3.12.1 to 3.20.0

mmastrac committed:
Merge pull request #112 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-checkstyle-plugin-3.5.0

Bump org.apache.maven.plugins:maven-checkstyle-plugin from 3.4.0 to 3.5.0

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-gpg-plugin from 3.2.5 to 3.2.6

Bumps [org.apache.maven.plugins:maven-gpg-plugin](https://github.com/apache/maven-gpg-plugin) from 3.2.5 to 3.2.6.
- [Release notes](https://github.com/apache/maven-gpg-plugin/releases)
- [Commits](https://github.com/apache/maven-gpg-plugin/compare/maven-gpg-plugin-3.2.5...maven-gpg-plugin-3.2.6)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-gpg-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #115 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-gpg-plugin-3.2.6

Bump org.apache.maven.plugins:maven-gpg-plugin from 3.2.5 to 3.2.6

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-gpg-plugin from 3.2.6 to 3.2.7

Bumps [org.apache.maven.plugins:maven-gpg-plugin](https://github.com/apache/maven-gpg-plugin) from 3.2.6 to 3.2.7.
- [Release notes](https://github.com/apache/maven-gpg-plugin/releases)
- [Commits](https://github.com/apache/maven-gpg-plugin/compare/maven-gpg-plugin-3.2.6...maven-gpg-plugin-3.2.7)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-gpg-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #116 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-gpg-plugin-3.2.7

Bump org.apache.maven.plugins:maven-gpg-plugin from 3.2.6 to 3.2.7

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.10.0 to 3.10.1

Bumps [org.apache.maven.plugins:maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.10.0 to 3.10.1.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.10.0...maven-javadoc-plugin-3.10.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-surefire-report-plugin

Bumps [org.apache.maven.plugins:maven-surefire-report-plugin](https://github.com/apache/maven-surefire) from 3.5.0 to 3.5.1.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.5.0...surefire-3.5.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-report-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.codehaus.mojo:exec-maven-plugin from 3.4.1 to 3.5.0

Bumps [org.codehaus.mojo:exec-maven-plugin](https://github.com/mojohaus/exec-maven-plugin) from 3.4.1 to 3.5.0.
- [Release notes](https://github.com/mojohaus/exec-maven-plugin/releases)
- [Commits](https://github.com/mojohaus/exec-maven-plugin/compare/3.4.1...3.5.0)

---
updated-dependencies:
- dependency-name: org.codehaus.mojo:exec-maven-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-site-plugin from 3.20.0 to 3.21.0

Bumps [org.apache.maven.plugins:maven-site-plugin](https://github.com/apache/maven-site-plugin) from 3.20.0 to 3.21.0.
- [Release notes](https://github.com/apache/maven-site-plugin/releases)
- [Commits](https://github.com/apache/maven-site-plugin/compare/maven-site-plugin-3.20.0...maven-site-plugin-3.21.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-site-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-checkstyle-plugin

Bumps [org.apache.maven.plugins:maven-checkstyle-plugin](https://github.com/apache/maven-checkstyle-plugin) from 3.5.0 to 3.6.0.
- [Commits](https://github.com/apache/maven-checkstyle-plugin/compare/maven-checkstyle-plugin-3.5.0...maven-checkstyle-plugin-3.6.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-checkstyle-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #121 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-checkstyle-plugin-3.6.0

Bump org.apache.maven.plugins:maven-checkstyle-plugin from 3.5.0 to 3.6.0

mmastrac committed:
Merge pull request #119 from mmastrac/dependabot/maven/org.codehaus.mojo-exec-maven-plugin-3.5.0

Bump org.codehaus.mojo:exec-maven-plugin from 3.4.1 to 3.5.0

mmastrac committed:
Merge pull request #120 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-site-plugin-3.21.0

Bump org.apache.maven.plugins:maven-site-plugin from 3.20.0 to 3.21.0

mmastrac committed:
Merge pull request #118 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-surefire-report-plugin-3.5.1

Bump org.apache.maven.plugins:maven-surefire-report-plugin from 3.5.0 to 3.5.1

mmastrac committed:
Merge pull request #117 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-javadoc-plugin-3.10.1

Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.10.0 to 3.10.1

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-surefire-report-plugin

Bumps [org.apache.maven.plugins:maven-surefire-report-plugin](https://github.com/apache/maven-surefire) from 3.5.1 to 3.5.2.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.5.1...surefire-3.5.2)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-report-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.10.1 to 3.11.2

Bumps [org.apache.maven.plugins:maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.10.1 to 3.11.2.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.10.1...maven-javadoc-plugin-3.11.2)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mmastrac committed:
Merge pull request #123 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-surefire-report-plugin-3.5.2

Bump org.apache.maven.plugins:maven-surefire-report-plugin from 3.5.1 to 3.5.2

mmastrac committed:
Merge pull request #124 from mmastrac/dependabot/maven/org.apache.maven.plugins-maven-javadoc-plugin-3.11.2

Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.10.1 to 3.11.2

heisluft committed:
Add Autoconversion of elegible Objects to Json (#125)

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-surefire-report-plugin (#132)

Bumps [org.apache.maven.plugins:maven-surefire-report-plugin](https://github.com/apache/maven-surefire) from 3.5.2 to 3.5.3.
- [Release notes](https://github.com/apache/maven-surefire/releases)
- [Commits](https://github.com/apache/maven-surefire/compare/surefire-3.5.2...surefire-3.5.3)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-surefire-report-plugin
  dependency-version: 3.5.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

NotJustAnna committed:
Fixes #98 by following modern Javascript's behaviour regarding NaN/Infinities. (#133)

stbischof committed:
Create JsonReader from Reader (#135)

Signed-off-by: Stefan Bischof <stbischof@bipolis.org>

mmastrac committed:
Use <release>8 instead of target/source (#138)

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-compiler-plugin (#127)

Bumps [org.apache.maven.plugins:maven-compiler-plugin](https://github.com/apache/maven-compiler-plugin) from 3.10.1 to 3.14.0.
- [Release notes](https://github.com/apache/maven-compiler-plugin/releases)
- [Commits](https://github.com/apache/maven-compiler-plugin/compare/maven-compiler-plugin-3.10.1...maven-compiler-plugin-3.14.0)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-compiler-plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

stbischof committed:
OSGi Support (#136)

- Removed maven-jar-plugin configuration.
- Added bnd-maven-plugin with execution goals.
- Configured Bundle-SymbolicName and Export-Package.
- Excluded unnecessary headers from the manifest.
- Improved JPMS module info generation.

Signed-off-by: Stefan Bischof <stbischof@bipolis.org>

stbischof committed:
Update test from junit4 to JUnit5 - BestPractices (#137)

Signed-off-by: Stefan Bischof <stbischof@bipolis.org>

mmastrac committed:
Bump versions

dependabot[bot] committed:
Bump org.codehaus.mojo:exec-maven-plugin from 3.5.0 to 3.5.1 (#139)

Bumps [org.codehaus.mojo:exec-maven-plugin](https://github.com/mojohaus/exec-maven-plugin) from 3.5.0 to 3.5.1.
- [Release notes](https://github.com/mojohaus/exec-maven-plugin/releases)
- [Commits](https://github.com/mojohaus/exec-maven-plugin/compare/3.5.0...3.5.1)

---
updated-dependencies:
- dependency-name: org.codehaus.mojo:exec-maven-plugin
  dependency-version: 3.5.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.junit.jupiter:junit-jupiter from 5.12.2 to 5.13.0 (#140)

Bumps [org.junit.jupiter:junit-jupiter](https://github.com/junit-team/junit5) from 5.12.2 to 5.13.0.
- [Release notes](https://github.com/junit-team/junit5/releases)
- [Commits](https://github.com/junit-team/junit5/compare/r5.12.2...r5.13.0)

---
updated-dependencies:
- dependency-name: org.junit.jupiter:junit-jupiter
  dependency-version: 5.13.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.junit.jupiter:junit-jupiter from 5.13.0 to 5.13.3 (#144)

Bumps [org.junit.jupiter:junit-jupiter](https://github.com/junit-team/junit-framework) from 5.13.0 to 5.13.3.
- [Release notes](https://github.com/junit-team/junit-framework/releases)
- [Commits](https://github.com/junit-team/junit-framework/compare/r5.13.0...r5.13.3)

---
updated-dependencies:
- dependency-name: org.junit.jupiter:junit-jupiter
  dependency-version: 5.13.3
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-gpg-plugin from 3.2.7 to 3.2.8 (#143)

Bumps [org.apache.maven.plugins:maven-gpg-plugin](https://github.com/apache/maven-gpg-plugin) from 3.2.7 to 3.2.8.
- [Release notes](https://github.com/apache/maven-gpg-plugin/releases)
- [Commits](https://github.com/apache/maven-gpg-plugin/compare/maven-gpg-plugin-3.2.7...maven-gpg-plugin-3.2.8)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-gpg-plugin
  dependency-version: 3.2.8
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

FireMasterK committed:
Buffer Pool, UTF Surrogate and test fixes.

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-javadoc-plugin from 3.11.2 to 3.11.3 (#146)

Bumps [org.apache.maven.plugins:maven-javadoc-plugin](https://github.com/apache/maven-javadoc-plugin) from 3.11.2 to 3.11.3.
- [Release notes](https://github.com/apache/maven-javadoc-plugin/releases)
- [Commits](https://github.com/apache/maven-javadoc-plugin/compare/maven-javadoc-plugin-3.11.2...maven-javadoc-plugin-3.11.3)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-javadoc-plugin
  dependency-version: 3.11.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.junit.jupiter:junit-jupiter from 5.13.3 to 5.13.4 (#145)

Bumps [org.junit.jupiter:junit-jupiter](https://github.com/junit-team/junit-framework) from 5.13.3 to 5.13.4.
- [Release notes](https://github.com/junit-team/junit-framework/releases)
- [Commits](https://github.com/junit-team/junit-framework/compare/r5.13.3...r5.13.4)

---
updated-dependencies:
- dependency-name: org.junit.jupiter:junit-jupiter
  dependency-version: 5.13.4
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Stypox committed:
Revert "Fix issue for semi-strings as well"

This reverts commit f2bd86667bb685c300366ec79f8e87f151ef4a94.

Stypox committed:
Revert "Fix Checkstyle errors"

This reverts commit b2c9c6f914dfd728f4e0f55038681054e091ab88.

Stypox committed:
Revert "Disallow semi-strings as value"

This reverts commit e68b36a2f2494082f2f580cac483ec1c55a7fe38.

Stypox committed:
Revert "Fix some more semi-string parsing bugs"

This reverts commit b2f4abdb4a421242234b7379fe50c8eab7c831a6.

Stypox committed:
Revert "Support keywords as keys"

This reverts commit 93da2aaba67dc78f4e43971e6ae51f81a5f7eb52.

Stypox committed:
Revert "Support single quote strings"

This reverts commit 3324e15a2286a942a658cb4023119920ebcfa163.

Stypox committed:
Revert "Support keys starting with a non-ASCII character"

This reverts commit 216af6347042193672b60a8c2d203f3ce9f94d33.

Stypox committed:
Revert "Support JS-like JSON"

This reverts commit 0363134ca32e5e9321a8f0f836d1cb2d449a5cf7.

Stypox committed:
Update README

Stypox committed:
Merge remote-tracking branch 'mmastrac/master' into revert-our-changes-and-use-firemasterk-2

Stypox committed:
Fix checkstyle

FireMasterK commented:
Nice, however in NewPipeExtractor you will need to add support for `LazyString`s, like this: https://github.com/FireMasterK/NewPipeExtractor/commit/165b459f115ed637530baf6e33f7f95b8b4887bd

Stypox commented:
@TobiGr the difference seems to be just 664 bytes which is totally fine in my opinion:
```
34626  nanojson-1.11-SNAPSHOT_with_fastdoubleparser.jar
33962  nanojson-1.11-SNAPSHOT_without_fastdoubleparser.jar
```

TobiGr reviewed (approved):
Tests pass locally and the commits were chosen reasonably.

Stypox committed:
Maintain compatibility with Java 11

Stypox commented:
I removed the commit that updates to Java 17, and pushed another commit for the `pom.xml` that makes sure that, independently of the JDK used during build, the resulting JAR is compatible with Java 11. There was only one small thing in the whole codebase that was incompatible with Java 11, so I fixed that.
